src/lightning_fabric/strategies/strategy.py

```diff
@@ -216,21 +216,22 @@ def reduce_boolean_decision(self, decision: bool, all: bool = True) -> bool:
         return decision
 
     def save_checkpoint(
-        self, checkpoint: Dict[str, Any], filepath: _PATH, storage_options: Optional[Any] = None
+        self, path: _PATH, state: Dict[str, Union[Module, Optimizer, Any]], storage_options: Optional[Any] = None
     ) -> None:
-        """Save model/training states as a checkpoint file through state-dump and file-write.
+        """Save model, optimizer, and other state as a checkpoint file.
 
         Args:
-            checkpoint: dict containing model and trainer state
-            filepath: write-target file's path
-            storage_options: parameter for how to save to storage, passed to ``CheckpointIO`` plugin
+            path: A path to where the file(s) should be saved
+            state: A dictionary with contents to be saved. If the dict contains modules or optimizers, their
+                state-dict will be retrieved and converted automatically.
+            storage_options: Additional options for the ``CheckpointIO`` plugin
         """
+        state = self._convert_stateful_objects_in_state(state)
```

Comment:
For now, I have given the strategy the responsibility to handle the objects in the state. This is to make the deepspeed implementation easier, since it needs access to everything.

Reply:
I think as this is mainly strategy-dependent we could also leave it that way.